src/collect/cli.rs

```diff
@@ -187,8 +187,13 @@ impl SubCommand for Collect {
 
         // Determine all registerd collectors and specify both the possible values and the default
         // value of the "collectors" argument
-        let possible_collectors =
-            Vec::from_iter(self.collectors.modules().iter().map(|x| x.to_str()));
+        let mut modules = crate::get_modules()?;
+        let collectors = modules.collectors();
+        let _ = collectors
+            .values()
+            .try_for_each(|c| c.register_cli(&mut self.collectors));
```

Comment:
I'm sorry, was about to merge this but only see this now. We should not silence the error here, please instead do:

```rust
collectors
            .values()
            .try_for_each(|c| c.register_cli(&mut self.collectors))?;
```

Reply:
Hmm, got error after update
```bash
# ./target/debug/retis collect -h
thread 'main' panicked at src/cli/cli.rs:351:80:
called `Result::unwrap()` on an `Err` value: module SkbDrop already registered
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Need to figure out how to avoid the module already registered issue.

Reply:
Ah right, we have this two step registration for the cli. Since we don't do this anymore I'm wondering if this is still needed (to do the second one). The following removes the second registration (untested):
```diff
diff --git a/src/cli/cli.rs b/src/cli/cli.rs
index 1d5ebd50c372..f2c76bd98a36 100644
--- a/src/cli/cli.rs
+++ b/src/cli/cli.rs
@@ -345,11 +345,6 @@ impl FullCli {
                 .join(" ")
         );
 
-        // Replace the ran subcommand with the full subcommand.
-        self.command = self
-            .command
-            .mut_subcommand(self.subcommand.name(), |_| self.subcommand.full().unwrap());
-
         // Get the matches.
         let matches = match cfg!(test) {
             true => self.command.try_get_matches_from_mut(self.args.iter())?,
```

Reply:
Thanks! I just tested it and works!